test/functional/helpers.lua

```diff
@@ -878,9 +878,11 @@ function module.os_kill(pid)
     or  'kill -9 '..pid..' > /dev/null'))
 end
 
--- Create directories with non exsisting intermidiate directories
+-- Create folder with non existing parents
 function module.mkdir_p(path)
-  return module.meths.call_function('mkdir', {path, 'p'})
+  return os.execute((iswin()
+    and 'mkdir '..path
+    or 'mkdir -p '..path))
```

Comment:
https://github.com/neovim/neovim/pull/14686#issuecomment-858937185

> I don't love that the new `mkdir_p` uses the current neovim session. I think it should use an os command (like `os_kill` or `do_rmdir`).

@tjdevries why? shelling out means we are solving platform/shell issues in our tests. Using the Nvim functions allows us to rely on the more robust implementation. Also, eventually we want to use a Nvim instance to run the tests, instead of a stock `lua` binary. So when the happens, we have to / should undo all of these places where we are shelling out...